### PR TITLE
feat: add whatauthorized

### PR DESCRIPTION
**Pull request by bakkelim**
## Description of change

Implement WhatAuthorized

### CODE REVIEW

**Review by arxeiss — APPROVED** [review]

File: functions/what_authorization.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2022 IndyKite
```
Comment (arxeiss) [inline functions/what_authorization.go:1]: 2023 ;)

File: functions/what_authorization_test.go [hunk 2]
```diff
@@ -0,0 +1,308 @@
+// Copyright (c) 2022 IndyKite
```
Comment (arxeiss) [inline functions/what_authorization_test.go:1]: 23 ;)

File: functions/what_authorization.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+			}
+			var resp *authorizationpb.WhatAuthorizedResponse
+			resp, err = client.WhatAuthorizedWithRawRequest(bCtx.Context, req)
+			var obj ast.Object
+
+			if statusErr := errors.FromError(err); statusErr != nil {
+				if errors.IsServiceError(statusErr) {
+					return nil, statusErr
+				}
+				obj = ast.NewObject(ast.Item(ast.StringTerm("error"), utilities.BuildUserError(statusErr)))
```
[the remark this answers is not in the document]
Comment (bakkelim) [reply functions/what_authorization.go:99]: Yeah, I need to take a dive into this once time allows

File: functions/what_authorization_test.go (hunk 2 above)
Comment (bakkelim) [reply functions/what_authorization_test.go:1]: added

**Pull request merged**